### ISSUE TITLE
add armv7l virtual machine

### ISSUE THREAD

**Issue by volth** [nix-community/aarch64-build-box#53]
It would be nice to run a 32-bit qemu virtual machine on the ```aarch64-build-box``` to perform ```aarch32``` (aka ```armv7l```) builds.

Unlike the old ThundeX2, this Hi1616 ```aarch64-build-box``` is able to run 32-bit code, but many build scripts of essencial software (including ```perl```, ```python```, ```gnutls```, ...) are not ready for that.
They took "aarch64" string out of ```$(uname -m)``` and then use ```aarch64``` assembler files to build ```aarch32``` software.

A 32-bit virtual machine which does not expose ```"aarch64"``` and pretends to be 32-bit only is an easy and promising solution for that

related:
 * Issue https://github.com/NixOS/nixpkgs/issues/60103 ```aarch32 stdenv should hide the architecture of the builder machine```
 * PR https://github.com/NixOS/nixpkgs/pull/60131 ```fix aarch32 build on aarch64 builder```
 * https://github.com/nix-community/aarch64-build-box/pull/39 ```Add armv7l-linux to extra-platforms```

cc @samueldr

**Comment by lheckemann** [comment]
It shouldn't be strictly necessary to make any changes to the system config for this I think, just info from @samueldr on how to set up such a VM.

**Comment by volth** [comment]
I suspect it will require an armv7 device or VM for initial bootstrap
(and to build the 32-bit qemu derivation)
So it would be convenient to have such a VM constantly running on the
build server

**Comment by volth** [comment]
```qemu``` can accept ```kernel``` and ```initrd``` as command-line arguments, it should know how to load and run them.
So ```grub``` and ```efi``` may be not needed actually

**Comment by samueldr** [comment]
Yes, definitely not _required_, though in my case it was the next step to boot EFI on real hardware :). It also was more convenient with regards to treating the VM like an installed machine, managed inside the VM, instead of being managed outside; the VM couldn't affect the qemu kernel and initrd parameters without much work.

**Comment by volth** [comment]
On the aarch64-build-box (Hi1616):

aarch64's ```qemu-system-arm -enable-kvm``` fails with
```
qemu-system-arm: -machine accel=kvm: No accelerator found
```
That's ok, the architectures are different

aarch32's ```qemu-system-arm -enable-kvm``` fails with
```
kvm version too old
qemu-system-arm: failed to initialize KVM: Invalid argument
```

**Comment by samueldr** [comment]
@volth check in `~samueldr/Public` (you should have access) I have uploaded my misc artifacts from the time I was working on that.

**Comment by volth** [comment]
ah, ```qemu-system-aarch64 -enable-kvm -cpu host,aarch64=off``` does the trick.

It is very promising: unlike ```qemu-system-arm```, ```qemu-system-aarch64``` has working ```9p```, so it should be possible to run an image-less VM (like those in ```test-driver``` and ```runInLinuxVM```)

**Comment by lheckemann** [comment]
related: https://github.com/grahamc/packet-nix-builder/pull/2
